core-rust/state-manager/src/staging/stage_tree.rs

```diff
@@ -218,36 +218,38 @@ impl<S: ReadableSubstateStore> StagedSubstateStoreManager<S> {
         &self.nodes.get(*key).unwrap().receipt
     }
 
-    fn recompute_data_recursive(
-        nodes: &mut SlotMap<StagedSubstateStoreNodeKey, StagedSubstateStoreNode>,
-        node_key: StagedSubstateStoreNodeKey,
-    ) {
-        let parent_store = ImmutableStore::from_parent(&nodes.get(node_key).unwrap().store);
-
-        let children_keys = nodes.get(node_key).unwrap().children_keys.clone();
-        for child_key in children_keys.iter() {
-            let child_node = nodes.get_mut(*child_key).unwrap();
-            child_node.store = parent_store.clone();
-            if let TransactionResult::Commit(commit) = &child_node.receipt.result {
-                commit.state_updates.commit(&mut child_node.store);
-            }
-            Self::recompute_data_recursive(nodes, *child_key);
-        }
-    }
-
     /// Rebuilds ImmutableStores by starting from the root with new, empty ones
     /// and recursively reapplies the [`receipt`]s.
     fn recompute_data(&mut self) {
         // Reset the [`dead_weight`]
         self.dead_weight = 0;
 
+        let mut stack = Vec::new();
```

Comment:
Do you think it might be worth a quick comment explaining why we have done it this way instead of recursively? TO prevent some future well-meaning soul refactoring it back into recursion? :')

(Ditto below?)